src/tools/illink/src/linker/Linker/TypeMapInfo.cs

```diff
@@ -128,15 +151,119 @@ protected virtual void MapType (TypeDefinition type)
 				MapType (nested);
 		}
 
+		void MapInterfacesOnType (TypeDefinition type)
+		{
+			if (_interfaces.ContainsKey (type))
+				return;
+
+			// Map from inflated interface type => list of every way to recursively implement that interface in 'type declaration order' according to ECMA 335 12.2
+			Dictionary<TypeReference, List<InterfaceImplementationNode>> waysToImplementIface = new (InflatedInterfaceComparer);
+
+			// Get all explicit interfaces of this type
+			foreach (var directIface in type.Interfaces) {
+				InterfaceImplementationNode directlyImplementedNode = new InterfaceImplementationNode (directIface, type, []);
+				TypeReference inflatedDirectIface = directIface.InterfaceType.TryInflateFrom (type, context)!;
+				waysToImplementIface.AddToList (inflatedDirectIface, directlyImplementedNode);
+			}
+
+			// Add interfaces on base type
+			if (type.BaseType is { } baseType && context.Resolve (baseType) is { } baseDef) {
+				MapInterfacesOnType (baseDef);
+				var baseInterfaces = _interfaces[baseDef];
+				foreach (var item in baseInterfaces) {
+					var inflatedInterface = item.InflatedInterface.TryInflateFrom (type.BaseType, context);
+					Debug.Assert (inflatedInterface is not null);
+					foreach (var node in item.InterfaceImplementationNodes) {
+						waysToImplementIface.AddToList (inflatedInterface, node);
```

Comment:
I can't think of a time that wouldn't work for marking methods, but I think it's valuable to keep the recursive interfaces list accurate and be explicit about keeping it for the interface rather than relying on processing in other places to keep it. I'll look into whether the perf difference would be significant, but if it's not I'd rather keep this as is.